edx-solutions/xblock-drag-and-drop-v2
# [MCKIN-7389-js-i18n] Adding i18n support in JS files

This PR adds support of i18n in Javascript of the DnDv2 xblock. Previously, the translation strings present in text.po and text.mo were not used for strings in JS. WIth the use of namespaced catalog files generated by [django-statici18n](https://github.com/zyegfryed/django-statici18n/pull/41). the translated strings are available by using the function window.DragAndDropI18N.gettext and window.DragAndDropI18N.ngettext

**JIRA tickets**: OSPR ticket

**Screenshots**: 
LMS view:
<img width="477" alt="screen shot 2018-05-17 at 12 49 20 pm" src="https://user-images.githubusercontent.com/9654006/40163732-08c5eafa-59d1-11e8-8e6b-fc216828b680.png">

Studio / Editing Views:
<img width="940" alt="screen shot 2018-05-17 at 12 49 54 pm" src="https://user-images.githubusercontent.com/9654006/40163749-14c886b4-59d1-11e8-9582-2555844d96bf.png">
<img width="416" alt="screen shot 2018-05-17 at 12 50 33 pm" src="https://user-images.githubusercontent.com/9654006/40163768-2057530c-59d1-11e8-9cfc-00809fec842e.png">
<img width="604" alt="screen shot 2018-05-17 at 12 51 01 pm" src="https://user-images.githubusercontent.com/9654006/40163774-24830cc8-59d1-11e8-826e-0d73b6cab2c4.png">


**Sandbox URL**: https://studio-pr18208.sandbox.opencraft.hosting/


**Testing instructions**:
1. As staff on a course, add the Drag and Drop problem to a course. It has already been added [here:](https://studio-pr18208.sandbox.opencraft.hosting/container/block-v1:OpenCraft+DnDv2+2018_1+type@vertical+block@ec13f23634824b94a6ddbf4d92afd244)
2. Change the language by visiting the [update_lang](https://studio-pr18208.sandbox.opencraft.hosting/update_lang/) URL. enter 'eo' (without quotes) for language code and click the submit button.
3. Visit the [DnDv2 problem](https://studio-pr18208.sandbox.opencraft.hosting/container/block-v1:OpenCraft+DnDv2+2018_1+type@vertical+block@ec13f23634824b94a6ddbf4d92afd244) page again and refresh the page. All strings should appear translated.
4. Click the Edit button to edit the settings of the DnDv2 problem. All strings should be appear translated.

**Author notes and concerns**:

1. Some strings have not been extracted into the .po/.mo files yet, and so they appear un-translated. They will be marked and translated in another iteration. Exporting missing translation strings is out of the scope of this PR.

2. Any text stored against an XBlocks won't be translated because it's considered "content". These initial feedback strings noted here are provided as suggestions to the educator, and can only be presented exactly as specified by the educator.


**Reviewers**
